### PR TITLE
python-typing-extensions 4.3.0 (new formula)

### PR DESCRIPTION
**Pull request by singingwolfboy**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

I'm adding this formula to resolve the build problems in #110199.

### CODE REVIEW

File: Formula/python-typing-extensions.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        .map(&:to_formula)
+        .sort_by(&:version)
+  end
+
+  def install
+    system Formula["flit"].opt_bin/"flit", "build", "--format", "wheel"
+    wheel = Pathname.glob("dist/typing_extensions-*.whl").first
+    pip_flags = %W[
+      -v
+      --no-deps
```
Comment (carlocab) [inline Formula/python-typing-extensions.rb:24]:
```suggestion
      --verbose
      --isolated
      --no-deps
```
Maybe? We definitely want to spell out `-v` as `--verbose` for ease of reading, anyway.

**Comment by SMillerDev** [comment]
If it's just a library I'm not sure it belongs in Homebrew.

**Comment by singingwolfboy** [comment]
> If it's just a library I'm not sure it belongs in Homebrew.

Is there a standard test or guideline for what belongs in Homebrew and what doesn't? `six` and `pyyaml` are both Python libraries, but they have formulae in Homebrew.

Also, as I mentioned in the pull request description, I created this formula to attempt to address a linkage problem with another formula: https://github.com/Homebrew/homebrew-core/pull/110199#issuecomment-1242925164 If `typing-extensions` was its own Homebrew formula, then other formulae that depend on it could declare the dependency in the formula, and avoid linkage conflicts. If you have a better suggestion for how to resolve this, I'm open to it!

**Comment by carlocab** [comment]
Python libraries that can be used in multiple formulae are ok by me.

**Comment by iMichka** [comment]
`six`, `pyaml`, `tabulate`, `pillow` are already accepted exceptions in homebrew.
I'm fine to add `python-typing-extensions` because it is a quite similar situation.

The only condition is that these formulae need to have Python 3.8, 3.9 and 3.10 as build and test dependencies, to be able to use them in a mixed Python dependency tree and not block any new Python migration.

We might need to update the documentation though:
https://docs.brew.sh/Formula-Cookbook#specifying-gems-python-modules-go-projects-etc-as-dependencies

We should also define a better rule for these exceptions in the documentation. Else there are many Python resources that could be added as formula.

**Comment by carlocab** [comment]
> The only condition is that these formulae need to have Python 3.8, 3.9 and 3.10 as build and test dependencies, to be able to use them in a mixed Python dependency tree and not block any new Python migration.

Mixed Python versions in the dep tree will no longer fail CI. See

https://github.com/Homebrew/homebrew-core/blob/c1def0c480b8011ae291984e8f6e2d769aa986a9/audit_exceptions/versioned_formula_dependent_conflicts_allowlist.json#L2

**Comment by iMichka** [comment]
Right sorry forgot that changed lately. This should be:

> The only condition is that these formulae need to have Python 3.8, 3.9 and 3.10 as build and test dependencies, to not block any new Python migration.

I am still wondering where we draw the line between resources and formula for these cases.

**Comment by singingwolfboy** [comment]
So where do we stand on this pull request? Can it be merged?

**Comment by carlocab** [comment]
Can we try a better test beyond just `import`?

**Comment by singingwolfboy** [comment]
> Can we try a better test beyond just `import`?

Good point. I've added a test using `mypy` to check that the type annotations work correctly.

**Comment by singingwolfboy** [comment]
@carlocab Is this ready to be merged?

**Comment by singingwolfboy** [comment]
Hi @carlocab, sorry to ping you again, but I don't want this to get forgotten about. Is there anything preventing this from being merged?

**Review by SMillerDev — APPROVED** [review]

**Review by chenrui333 — APPROVED** [review]

**Comment by singingwolfboy** [comment]
Could someone please merge this pull request? @carlocab? @SMillerDev? @chenrui333? Or if there's something more I need to do, please let me know, but it doesn't look that way to me.

**Comment by carlocab** [comment]
Sorry, too many PRs to look at there. Should be merged soon.

**Pull request closed without merging**